scwx-qt/source/scwx/qt/model/marker_model.cpp

```diff
@@ -297,14 +297,19 @@ void MarkerModel::HandleMarkerAdded(types::MarkerId id)
    const int newIndex = static_cast<int>(*index);
```

Comment:
index is optional, but this assumes it always has a value. Since `HandleMarkerAdded` is a slot, this could be a queued/deferred call. It's probably safe to just skip this processing if index is empty, but should have some checking regardless to prevent undefined behavior.